### ISSUE TITLE
Global Navigation Menu

### ISSUE THREAD

**Issue by martypowell** [Baltimore-County-Government/dotgov-components#77]
## Component Name

GlobalNavigation (not loving this)

## Description

A hamburger menu approach which will display links for users to navigate the site. The navigation will slide from left to right, and push the page content to the right.

**UPDATE**: A gray transparent overlay will be included when expanded to disabled the rest of site expect the menu. If you select that area, the menu should collapse and the overlay should be removed, shifting the website back to it's normal position.

## Required and optional properties

- List of navigation links

### Examples of Expanded and Collapsed states

**Collapsed**

![image](https://user-images.githubusercontent.com/1933400/63041408-d2f22580-be95-11e9-9c33-cab638f113ee.png)

**Expanded**

![image](https://user-images.githubusercontent.com/1933400/63041433-e4d3c880-be95-11e9-8474-428841c1770d.png)

### Full Examples

See the [design document](https://xd.adobe.com/spec/6c7ed8ba-6428-418d-464e-d31fb3608c6b-1a4f/screen/8549b380-ac64-46be-87bf-a4b71b38cee6/10-Navigation/) for complete style detail.

## Acceptance Criteria

Please include in the following format. In some cases you will not need both types.

**Rule Based**:

- Must follow UX standards
- Semantically correct - Must follow [W3C](https://www.w3.org/) standards
- Matches Mockup Styles

### Scenario Based

**Scenario**: Site Menu is collapsed
**Given**: A user sees the menu button at the top left of the screen
**When**: User selects the button
**Then**: The menu icon is changed to an 'X', the navigation slides in from the left, and the user is able to view the navigation menu, and the main contents of the site, are covered by a transparent gray background.

**Scenario**: Site Menu is expanded
**Given**: A user wants to navigate to a menu item
**When**: User selects the menu item
**Then**: The user is taken to menu items page

**Scenario**: Site Menu is expanded
**Given**: A user wants to collapse the menu
**When**: User selects the menu item (X) or clicks a non-link element anywhere on the page canvas.
**Then**: The menu closes from right to left, the menu icon returns to the triple bar status, the navigation is no longer visible. The transparent gray background is no longer visible.

## Additional context

n/a

**Comment by martypowell** [comment]
@theilman7 @danfox01 requirements updated to include a gray transparent background to "disable" the website apart from the menu, when the menu is expanded.

An example of this can be found at mobile sizes for bcpl https://www.bcpl.info/blog

**Comment by theilman7** [comment]
@danfox01 @martypowell Acceptance criteria on this entry may need to be updated as this should no longer animate to an X on-click, correct?

**Comment by martypowell** [comment]
Yes @theilman7 for now, no animation on the icons, but the acceptance criteria is ok, the bars will still change to an x.

**Comment by martypowell** [comment]
@danfox01 @theilman7 this is moved to the last priority to ensure that we get everything else done.
